dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -140,7 +138,6 @@ func (rrh *resolverRequestHandler) requestHashesWithDataSplit(
 			"num txs", len(unrequestedHashes),
 			"max txs to request", rrh.maxTxsToRequest,
 		)
-		return err
```

Comment:
You can add return here

Reply:
would leave as it is - the next for could do nothing as is a range over a nil.